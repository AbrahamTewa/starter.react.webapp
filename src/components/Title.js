import React from 'react';
import PropTypes from 'prop-types';

/**
 * Title of the application
 * @param {string} children
 * @returns {XML}
 * @constructor
 */
function Title({ children }) {
    return <h1>{children}</h1>;
}

Title.propTypes = { children: PropTypes.string.isRequired };

export default Title;

