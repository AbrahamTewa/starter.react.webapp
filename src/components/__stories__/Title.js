/* eslint-env node */

import React from 'react';
import { storiesOf } from '@storybook/react';

import Title from '../Title';

storiesOf('Title', module)
    .add('without text', () => (
        <Title/>
    ))
    .add('with text', () => (
        <Title>tox</Title>
    ));
