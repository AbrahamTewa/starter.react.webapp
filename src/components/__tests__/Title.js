/* eslint-env node, jest */

import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import Title from '../Title';

describe('Basic tests', () => {
    test('Emtpy string', () => {
        let render;
        let shallowComponent;
        let title;

        title = <Title/>;

        // Snapshot
        render           = renderer.create(title);
        expect(render.toJSON()).toMatchSnapshot();

        // Enzyme
        shallowComponent = shallow(title);
        expect(shallowComponent.text()).toEqual('');
    });

    test('Simple string', () => {
        let render;
        let shallowComponent;
        let title;
        let titleText;

        titleText = 'One simple string';
        title = <Title>{titleText}</Title>;

        // Snapshot
        render = renderer.create(title);
        expect(render.toJSON()).toMatchSnapshot();

        // Enzyme
        shallowComponent = shallow(title);
        expect(shallowComponent.text()).toEqual(titleText);
    });
});
