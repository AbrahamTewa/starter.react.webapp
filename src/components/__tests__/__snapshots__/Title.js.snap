// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Emtpy string 1`] = `<h1 />`;

exports[`Basic tests Simple string 1`] = `
<h1>
  One simple string
</h1>
`;
